subprojects/build-plugins/build-plugins.gradle.kts

```diff
@@ -0,0 +1,55 @@
+import build.*
+
+plugins {
+    id("public-kotlin-dsl-module")
+}
+
+base {
+    archivesBaseName = "gradle-kotlin-dsl-build-plugins"
+}
+
+
+val processResources: ProcessResources by tasks
+val writeKotlinDslProviderVersion by tasks.creating(WriteProperties::class) {
+    outputFile = processResources.destinationDir.resolve("${base.archivesBaseName}-versions.properties")
+    property("kotlin-dsl", version)
+}
+processResources.dependsOn(writeKotlinDslProviderVersion)
+
+
+val writeTestKitPluginClasspath by tasks.creating {
```

Comment:
Why not applying the [java gradle [dev] plugin](https://docs.gradle.org/current/userguide/java_gradle_plugin.html)? 🤔